src/cargo/core/interning.rs

```diff
@@ -0,0 +1,67 @@
+use std::fmt;
+use std::sync::RwLock;
+use std::collections::HashSet;
+use std::slice;
+use std::str;
+use std::mem;
+use std::cmp::Ordering;
+
+pub fn leek(s: String) -> &'static str {
```

Comment:
One thing this may actually want to do is call `s.into_boxed_str()` as that'll call `shink_to_fit` and make sure we're not hanging on to any lingering bytes, but that can happen in a follow-up!

Reply:
```
    let boxed = s.into_boxed_str();
    let ptr = boxed.as_ptr();
    let len = boxed.len();
    mem::forget(boxed);
```
Yes?

Reply:
indeed!

Eventually this could even use `Box::leak` :)